### DISCUSSION TITLE
Handling non-compliant HAR files?

### DISCUSSION THREAD

**Discussion by FineFindus** [mandrean/har-rs#28] [Ideas]
Reading the following [.har file](https://github.com/mandrean/har-rs/files/10538766/github.com_Archive.23-01-30.19-18-53.zip) fails with the error message `error serializing YAML`, even though the file is not a YAML file.

I believe this error is specific to Firefox, as it does not appear when generating `.har` files with chromium based browser.

Small example code:
```rust
fn main() {
    match har::from_path("path of input") {
        Ok(spec) => {}
        Err(err) => {
            println!("Failed to read file: {}", err);
        }
    }
}
```

**Comment by mandrean** [comment]
Hi @FineFindus, thanks for filing an issue

According to the [HAR spec](http://www.softwareishard.com/blog/har-12-spec/), the `lastAccess` field is **required**:
![image](https://user-images.githubusercontent.com/4631744/228099300-47ded451-444d-4ea3-b2cd-3448410e1eee.png)

But its not there in the HAR archive in your link, hence it throws the error `missing field lastAccess`.

Maybe "modes" could be implemented, something like `STRICT` and `ALLOW_MISSING_FIELDS`. I'll have to think about it

**Edit:** updated the title & converted it to a discussion

**Comment by FineFindus** [comment]
> According to the [HAR spec](http://www.softwareishard.com/blog/har-12-spec/), the lastAccess field is required:

Interesting, makes sense why it failed then. In that case, the issue should also be moved to the Firefox repository. I opened it here, as I was able to read it with another deserializer and did not expect it to be Firefox's fault. Also, it did not throw the `missing field lastAccess` error for me, is that a new addition?

I think it would be a great idea to add different modes, as other browser might also create corrupted files from time to time, which would otherwise not be readable with the crate.
